app/views/ErrorMapper.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import play.api.data.FormError
+
+object ErrorMapper {
+
+  def eoriOrAddressErrors(groupKey: String, errors: Seq[FormError]) =
+    errors.map(err => err.copy(key = if (err.key == groupKey) s"$groupKey.eori" else err.key))
+
+  def radioGroupErrors(groupKey: String, firstOption: String, errors: Seq[FormError]) =
```

Comment:
This is for "general" radio groups.
The "yesNo" version is for radio groups using the pre-built 'yesNoRadios' component

Reply:
Do you think it is feasable to have `Mapping` type instead of `firstOption` and then taking the first field name from it? I am not sure if it is ordered or accessible in any way. But this could save someone from little errors with order or even a typo.
In a perfect world you could create an extension for `Form` or `FormError` classes to integrate with existing code. But that is probably not worth the effort.

Reply:
The "firstOption" is the id of the first valid radio button.   It's not in the `mapping`